### PR TITLE
Api updated to v2

### PR DESCRIPTION
**Pull request by ProcyonLotor123**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204464594667832

### CODE REVIEW

File: src/swap/exolix.js
Comment (paullinator) [inline src/swap/exolix.js:154]: You don't differentiate between the from and to amounts. If `request.quoteFor === 'to'` then the user is specifying the amount of crypto to receive and will be sending the `fromMainnetCode` currency. The code below just assumes from and to are in reverse. Major bug! Bug exists in current plugin which I'll have to fix for now to not allow reverse quotes.

**Review by paullinator — CHANGES_REQUESTED** [review]

**Comment by paullinator** [comment]
PR needs to be rebased on latest master (DO NOT merge master into your feature branch). Latest version already includes better use of cleaners and TS types.

File: src/swap/exolix.ts
Comment (paullinator) [inline src/swap/exolix.ts:156]: This is not a `native` amount. Should be renamed to `exchangeQuoteAmount` "exchange..." refers to denominations in the usual amount used by exchange. ie BTC, ETH not mETH, sats, bits.

File: src/swap/exolix.ts
Comment (paullinator) [inline src/swap/exolix.ts:185]: Both of these `quoteParams` are identical. You need to differentiate FROM vs TO quotes.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/swap/exolix.ts
Comment (devexolix) [reply src/swap/exolix.ts:185]: What do you mean? 
On line 156 we received `exchangeQuoteAmount` from `request.quoteFor = from|to` And if `request.quoteFor === 'to'` we will receive quoteAmount differen with quoteAmount as `request.quoteFor === 'from'`

File: src/swap/exolix.ts
Comment (devexolix) [reply src/swap/exolix.ts]: fixed

File: src/swap/exolix.ts
Comment (devexolix) [reply src/swap/exolix.ts:185]: @paullinator Do we need any more updates from our side?

**Review by devexolix — APPROVED** [review]

File: src/swap/exolix.ts
Comment (Jon-edge) [inline src/swap/exolix.ts:164]: We leave the same `quoteAmount` in `amount` for a `to` quote?

Comment (ProcyonLotor123) [reply src/swap/exolix.ts:164]: Yes, `withdrawalAmount` has a priority over `amount` and we can leave it.

**Review by Jon-edge — APPROVED** [review]

File: src/swap/exolix.ts
Comment (paullinator) [inline src/swap/exolix.ts:209]: you already check minimums above. This logic now looks incorrect for 'to' quotes assuming minAmount is always the 'from' amount

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/swap/exolix.ts
Comment (aguraan) [reply src/swap/exolix.ts:197]: @paullinator Thanks for noticing, corrected.

Comment (ProcyonLotor123) [reply src/swap/exolix.ts:197]: @paullinator Thanks for noticing, corrected.

**Review by paullinator — APPROVED** [review]
Please squash all into one commit before merge

**Pull request merged**
